autoload/clap/themes/srcery.vim

```diff
@@ -0,0 +1,137 @@
+" Author: Birger J. Nordolum  <contact+srcery@mindtooth.no>
+" Description: Clap theme based on the Srcery color scheme.  The theme
+" require some tweaks to make it work.  If you are to copy this code for
+" your own color scheme, or build upon it.  Make sure that you are aware
+" of the additional tweaks for Srcery.
+
+
+let s:save_cpo = &cpoptions
+set cpoptions&vim
+
+
+" Helper Functions {{{
+
+function! s:GetHightlightColor(group, type) abort
+	if a:type ==? 'fg'
+		let l:gui_color = synIDattr(hlID(a:group), 'fg', 'gui')
+		let l:term_color = synIDattr(hlID(a:group), 'fg', 'cterm')
+	elseif a:type ==? 'bg'
+		let l:gui_color = synIDattr(hlID(a:group), 'bg', 'gui')
+		let l:term_color = synIDattr(hlID(a:group), 'bg', 'cterm')
+	endif
+
+	return [ l:gui_color, l:term_color ]
+endfunction
+
+" }}}
+" Srcery Palette {{{
+
+let s:black          = ['#1C1B19', 0]
```

Comment:
is there any reason why these variables can get assigned by `GetHighlightColor`?
```viml
let s:black = s:GetHightlightColor('SrceryBlack', 'fg')

```

Reply:
> Think there's a typo here, shouldn't it be GetHighlightColor?

😆

Reply:
I don't think they are, but IIRC there were some reason for that...I might be wrong

Reply:
```viml
" Srcery Palette {{{

" Normal Colors
let s:black          = s:GetHighlightColor('SrceryBlack', 'fg')
let s:red            = s:GetHighlightColor('SrceryRed', 'fg')
let s:green          = s:GetHighlightColor('SrceryGreen', 'fg')
let s:yellow         = s:GetHighlightColor('SrceryYellow', 'fg')
let s:blue           = s:GetHighlightColor('SrceryBlue', 'fg')
let s:magenta        = s:GetHighlightColor('SrceryMagenta', 'fg')
let s:cyan           = s:GetHighlightColor('SrceryCyan', 'fg')
let s:white          = s:GetHighlightColor('SrceryWhite', 'fg')

" Bright Colors
let s:bright_black   = s:GetHighlightColor('SrceryBrightBlack', 'fg')
let s:bright_red     = s:GetHighlightColor('SrceryBrightRed', 'fg')
let s:bright_green   = s:GetHighlightColor('SrceryBrightGreen', 'fg')
let s:bright_yellow  = s:GetHighlightColor('SrceryBrightYellow', 'fg')
let s:bright_blue    = s:GetHighlightColor('SrceryBrightBlue', 'fg')
let s:bright_magenta = s:GetHighlightColor('SrceryBrightMagenta', 'fg')
let s:bright_cyan    = s:GetHighlightColor('SrceryBrightCyan', 'fg')
let s:bright_white   = s:GetHighlightColor('SrceryBrightWhite', 'fg')

" Extra Colors
let s:orange         = s:GetHighlightColor('SrceryOrange', 'fg')
let s:bright_orange  = s:GetHighlightColor('SrceryBrightOrange', 'fg')
let s:hard_black     = s:GetHighlightColor('SrceryHardBlack', 'fg')
let s:xgray1         = s:GetHighlightColor('SrceryXgray1', 'fg')
let s:xgray2         = s:GetHighlightColor('SrceryXgray2', 'fg')
let s:xgray3         = s:GetHighlightColor('SrceryXgray3', 'fg')
let s:xgray4         = s:GetHighlightColor('SrceryXgray4', 'fg')
let s:xgray5         = s:GetHighlightColor('SrceryXgray5', 'fg')
let s:xgray6         = s:GetHighlightColor('SrceryXgray6', 'fg')

"}}}
```

Guess we can use this other places too.

Reply:
Nice, yeah takes us one step close to a single truth.

Reply:
Just saw that you already did this in the lightline scheme. 😅

Added it to airline as well.  Care to look?

Maybe we could streamline it by adding a single file in e.g. `/autoload/srcery/helper.vim` and then define it as `function! g#srcery#helper#get_color()` or something?

Reply:
Looks good. Yeah, I'm all for setting up a common helper function, DRY and all that :)

Reply:
You wanna include that in this pull request or setup a separate one?

Reply:
Separate one for now so we can merge this.